runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -2382,9 +2382,23 @@ class INTERPRETER_CLASS
 #endif /* FFI_NATIVE_RAW_API */
 		}
 		{
-			args[0] = &ffi_type_pointer;
+#if defined(J9VM_ZOS_3164_INTEROPERABILITY)
+			if (J9_ARE_ALL_BITS_SET((UDATA)function, J9PORT_SL_ZOS_31BIT_TARGET_HIGHTAG)) {
+				/* For 31-bit cross-amode targets, we need to return the correponding
+				 * 31-bit JNIEnv* pointer instead.
+				 */
+				args[0] = &ffi_type_sint32;
+				if (NULL == _currentThread->jniEnv31) {
+					queryJNIEnv31(_currentThread);
+				}
+				values[0] = &(_currentThread->jniEnv31);
```

Comment:
Does this not require a `void*` cast?

Reply:
The compiler didn't complain, but I added the cast to `void*` for better consistency.  Fixed under: 48f9cc6c856c83d1079c8f8b9ce6bde733d9855b